pkg/pods/pod.go

```diff
@@ -593,6 +596,34 @@ func (pod *Pod) Launchables(manifest Manifest) ([]launch.Launchable, error) {
 	return launchables, nil
 }
 
+func (pod *Pod) SetFinishExec(finishExec []string) {
+	var runAs string
+
+	if manifest, err := pod.CurrentManifest(); err == nil {
+		runAs = manifest.RunAsUser()
```

Comment:
For now, and for the indefinite future, the `finish` program is only configurable by preparer config. That makes it safer than arbitrary user code - but yes, if they were already running as root, this problem exists no matter what.